### PR TITLE
add context timeout to long chain operations

### PR DESCRIPTION
**Pull request by jonasagx**
Fix for integration test timeout.

By adding context cancelation, to operations that _might_ get stuck, this PR fixes the timeout problem. The remaining question is what exactly is stuck? We might need more test coverage to handle that.

Signed-off-by: Jonas Xavier <jonasx@anchore.com>

### CODE REVIEW

File: test/cli/utils_test.go [hunk 1]
```diff
@@ -51,8 +54,20 @@ func setupPKI(t *testing.T, pw string) func() {
 
 func getFixtureImage(t testing.TB, fixtureImageName string) string {
 	t.Logf("obtaining fixture image for %s", fixtureImageName)
-	imagetest.GetFixtureImage(t, "docker-archive", fixtureImageName)
-	return imagetest.GetFixtureImageTarPath(t, fixtureImageName)
+	request := imagetest.PrepareFixtureImage(t, "docker-archive", fixtureImageName)
```
Comment (jonasagx) [inline test/cli/utils_test.go:57]: this code was copied from stereoscope so I can add a timeout, and prove the concept locally. It makes sense to change in stereoscope

**Comment by jonasagx** [comment]
Got an actual error now, yay!
```
=== RUN   TestPowerUserCmdFlags/content-cataloger-wired-up
[197](https://github.com/anchore/syft/runs/6342579912?check_suite_focus=true#step:10:197)
    power_user_cmd_test.go:98: data="\"contents\": \"c29tZV9BcEkta0V5ID0gIjEyMzQ1QTdhOTAxYjM0NTY3ODkwMTIzNDU2Nzg5MDEyMzQ1Njc4OTAxMjM0NTY3ODkwMTIzNDU2Nzg5MCIK\"" was NOT found in any output, but should have been there
[198](https://github.com/anchore/syft/runs/6342579912?check_suite_focus=true#step:10:198)
    power_user_cmd_test.go:98: expected no failure but got rc=-1
[199](https://github.com/anchore/syft/runs/6342579912?check_suite_focus=true#step:10:199)
    power_user_cmd_test.go:101: STDOUT:
[200](https://github.com/anchore/syft/runs/6342579912?check_suite_focus=true#step:10:200)
         
[201](https://github.com/anchore/syft/runs/6342579912?check_suite_focus=true#step:10:201)
    power_user_cmd_test.go:102: STDERR:
```

File: test/cli/utils_test.go [hunk 2]
```diff
@@ -85,7 +100,10 @@ func runSyftInDocker(t testing.TB, env map[string]string, image string, args ...
 }
 
 func runSyft(t testing.TB, env map[string]string, args ...string) (*exec.Cmd, string, string) {
-	cmd := getSyftCommand(t, args...)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	cmd := exec.CommandContext(ctx, getSyftBinaryLocation(t), args...)
```
Comment (kzantow) [inline test/cli/utils_test.go:105]: I wonder if we could detect the timeout and re-run with `-vv` ?

**Review by kzantow — APPROVED** [review]
I'm going to make the call to merge this for now, given we have broken tests and this seems to be at least some sort of workaround to make them pass reliably.

**Pull request merged**